sdks/python/apache_beam/io/textio_test.py

```diff
@@ -438,226 +432,241 @@ def test_read_all_many_file_patterns(self):
 
   def test_read_auto_bzip2(self):
     _, lines = write_data(15)
-    file_name = self._create_temp_file(suffix='.bz2')
-    with bz2.BZ2File(file_name, 'wb') as f:
-      f.write('\n'.join(lines))
-
-    pipeline = TestPipeline()
-    pcoll = pipeline | 'Read' >> ReadFromText(file_name)
-    assert_that(pcoll, equal_to(lines))
-    pipeline.run()
+    with TempDir() as tempdir:
```

Comment:
<!--new_thread; commit:5d08ca3beac2fd751bbfdcf83d47e57ce1411bc0; resolved:1-->
Thanks! this is much nicer.

Reply:
No problem! Hopefully this will be useful for more reader tests!